arkouda/pdarraycreation.py

```diff
@@ -472,23 +494,29 @@ def zeros(
     >>> ak.zeros(5, dtype=ak.bool_)
     array([False, False, False, False, False])
     """
-    if not np.isscalar(size):
-        raise TypeError(f"size must be a scalar, not {size.__class__.__name__}")
-    dtype_name = akdtype(dtype).name
+    dtype = akdtype(dtype)  # normalize dtype
+    dtype_name = dtype.name if isinstance(dtype, bigint) else cast(np.dtype, dtype).name
     # check dtype for error
     if dtype_name not in NumericDTypes:
-        raise TypeError(f"unsupported dtype {akdtype(dtype)}")
-    repMsg = generic_msg(cmd=f"create<{dtype_name},1>", args={"shape": size})
+        raise TypeError(f"unsupported dtype {dtype}")
+    shape, ndim, full_size = _infer_from_size(size)
+
+    if ndim > get_max_array_rank():
+        raise ValueError(f"array rank {ndim} exceeds maximum of {get_max_array_rank()}")
+
+    if ndim > 1:
+        return xp.zeros(shape, dtype=dtype)  # type: ignore
+    repMsg = generic_msg(cmd=f"create<{dtype_name},{ndim}>", args={"shape": shape})
```

Comment:
along the same lines, this block of code could just be:

```chapel
return create_pdarray(
  generic_msg(cmd=f"create<{dtype_name},{ndim}>", args={"shape": shape})
)
```

no `if ndim > 1` conditional needed, since the `create` command is already generic wrt `ndim`.

And then at some point, `zeros` in the array api module could/should be a wrapper around this function (instead of calling `f"create<{dtype_name},{ndim}>"` itself):

```chapel
def zeros(
    shape: Union[int, Tuple[int, ...]],
    /,
    *,
    dtype: Optional[Dtype] = None,
    device: Optional[Device] = None,
) -> Array:
    """
    Return a new array with the specified shape and type, filled with zeros.
    """
    from .array_object import Array

   ...

    if dtype is None:
       dtype = ak.Int64

    return Array._new(ak.zeros(shape, dtype))
```

Reply:
sounds good, definitely doesn't need to happen any time soon, but would be nice from a software maintainability standpoint.